src/leaf_intf.ml

```diff
@@ -15,6 +15,7 @@ module type S = sig
 
   val load : store -> address -> t
 
+  (* Lot's of duplication between [Node] and [Leaf], but I guess that's not that terrible. *)
```

Comment:
I don't like it either. I think both modules should go, and use directly `Vertex.LeafMake` and `Vertex.NodeMake`(there isn't really any logic implemented in `Leaf` and `Node`)